countWords.rb

```diff
@@ -0,0 +1,13 @@
+file = File.open("speech.txt", "r")
+content = file.read
+frequency = Hash.new(0)
+wordcount = content.split(" ")
+wordcount.each do |w|
+  w.gsub!(/\W|_/,"")
+  frequency[w] += 1
+end
+frequency = frequency.sort_by {|word,count| count}.reverse
+frequency.each do |key,value|
+  puts "#{key}:#{value}"
+end
+file.close
```

Comment:
This is a pretty good procedural approach and I think your approach is fine for solving this problem. However, just for fun (and because we have attributes and methods that act on those attributes), how about encapsulating this in a nice WordCounter class that accepts a file object on initialization and can be asked to print out a word count? :smiley: